proc/test_thread_rand.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Phoenix-RTOS
+ *
+ * phoenix-rtos-tests: random lifespan threads test
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/threads.h>
+
+#include "unity_fixture.h"
+
+
+typedef struct {
+	unsigned int time;
+	unsigned int id;
+} thread_params_t;
+
+
+typedef struct {
+	thread_params_t params;
+	char stack[512] __attribute__((aligned(8))); /* stack needs to be aligned to 8 */
+} thread_self_t;
+
+
+typedef struct {
+	unsigned int fin;
+	handle_t fin_lock;
+	thread_self_t threads[10]; /* set threads amount here */
+} threads_common_t;
+
+
+static threads_common_t threads_common;
+
+
+static void test_thread(void *arg)
+{
+	thread_params_t *params = (thread_params_t *)arg;
+	int err;
+	usleep(params->time * 1000);
+
+	err = mutexLock(threads_common.fin_lock);
+	TEST_ASSERT_EQUAL_INT(EOK, err);
+	threads_common.fin++;
+	err = mutexUnlock(threads_common.fin_lock);
+	TEST_ASSERT_EQUAL_INT(EOK, err);
+
+	endthread();
+}
+
+
+TEST_GROUP(threads_rand);
+
+
+TEST_SETUP(threads_rand)
+{
+}
+
+
+TEST_TEAR_DOWN(threads_rand)
+{
+}
+
+
+TEST(threads_rand, test_1)
+{
+	int cthread, err, i, joined = 0;
+
+	cthread = sizeof(threads_common.threads) / sizeof(threads_common.threads[0]);
+	err = mutexCreate(&(threads_common.fin_lock));
+	TEST_ASSERT_EQUAL_INT(EOK, err);
+	threads_common.fin = 0;
+
+	srand(7);
+	/* assign threads parameters */
+	for (i = 0; i < cthread; i++) {
+		threads_common.threads[i].params.id = i;
+		threads_common.threads[i].params.time = 100 + rand() % 200; /* assign thread lifespan in milliseconds */
```

Comment:
I've asked you to use RAND_MAX constant to scale the randomization space to defined by constant.

Reply:
Ok, added random number generation based on RAND_MAX and rand().

Also, for better randomization:
 - switched to microseconds
 - added `timemax` and `timemin` to `threads_common`